contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/plugin_subsystem_base.py

```diff
@@ -6,26 +6,50 @@
 
 import json
 
+from pants.option.parser import Parser
 from pants.subsystem.subsystem import Subsystem
 from pants.util.memo import memoized
 
+from pants.contrib.python.checks.checker.common import CheckstylePlugin
+
 
 class PluginSubsystemBase(Subsystem):
   @classmethod
   def plugin_type(cls):
+    """Returns the type of the plugin this subsystem configures."""
     raise NotImplementedError('Subclasses must override and return the plugin type the subsystem '
                               'configures.')
 
+  _option_names = None
+
   @classmethod
   def register_options(cls, register):
-    super(PluginSubsystemBase, cls).register_options(register)
+    option_names = []
+
+    def recording_register(*args, **kwargs):
+      option_names.append(Parser.parse_dest(*args, **kwargs))
+      register(*args, **kwargs)
+
+    super(PluginSubsystemBase, cls).register_options(recording_register)
+
     # All checks have this option.
-    register('--skip', type=bool,
-             help='If enabled, skip this style checker.')
+    recording_register('--skip', type=bool, help='If enabled, skip this style checker.')
+
+    cls.register_plugin_options(recording_register)
+    cls._option_names = frozenset(option_names)
+
+  @classmethod
+  def register_plugin_options(cls, register):
+    """Register options for the corresponding plugin."""
+    raise NotImplementedError('Subclasses must override instead of `register_options`.')
 
   def options_blob(self):
+    assert self._option_names is not None, (
+      'Expected `register_options` to be called before any attempt to read the `options_blob`.'
+    )
     options = self.get_options()
-    options_dict = {option: options.get(option) for option in options}
+    options_dict = {option: options.get(option)
```

Comment:
It would be useful to land this without another CI iteration, and it looks sane.

But I think that it might be possible to do this with `self.get_options().for_scope(.., inherit_from_enclosing_scope=False)`.

Reply:
I'm not seeing how to do this. It's `self.get_options() -> OptionValueContainer` and not `self.get_options() -> Options`.